src/main/java/ch/njol/skript/expressions/ExprColorOf.java

```diff
@@ -18,76 +18,140 @@
 
+	private static final boolean MAPS_AND_POTIONS_COLORS = Skript.methodExists(PotionMeta.class, "setColor", org.bukkit.Color.class);
+	private static final DyeColor DEFAULT_MATERIAL_COLOR = DyeColor.WHITE;
+	public static final Pattern MATERIAL_COLORS_PATTERN;
+
 	static {
-		register(ExprColorOf.class, Color.class, "colo[u]r[s]", "blocks/itemtypes/entities/fireworkeffects");
+		DyeColor[] dyeColors = DyeColor.values();
+		StringBuilder colors = new StringBuilder();
+		for (int i = 0; i < dyeColors.length; i++) {
+			colors.append(dyeColors[i].name()).append(i + 1 != dyeColors.length ? "|" : "");
+		}
+		MATERIAL_COLORS_PATTERN = Pattern.compile("^(" + colors + ")_.+");
+		register(ExprColorOf.class, Color.class, "colo[u]r[s]", "blocks/itemtypes/entities/fireworkeffects/slots");
 	}
 	
-	@SuppressWarnings("null")
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr(exprs[0]);
 		return true;
 	}
 	
-	@SuppressWarnings("null")
 	@Override
-	protected Color[] get(Event e, Object[] source) {
+	protected Color[] get(Event event, Object[] source) {
+		// TODO FIX
+		// this approach has couple issues, users can't use multiple types in source like
+		// 'broadcast colors of ((trailing burst firework colored blue and red) and targeted block)' and second
+		// this check doesn't work with variables as it's not converted yet
 		if (source instanceof FireworkEffect[]) {
 			List<Color> colors = new ArrayList<>();
-			
+
 			for (FireworkEffect effect : (FireworkEffect[]) source) {
 				effect.getColors().stream()
-					.map(SkriptColor::fromBukkitColor)
+					.map(ColorRGB::fromBukkitOrRgbColor)
 					.forEach(colors::add);
 			}
-			
+
 			if (colors.size() == 0)
 				return null;
 			return colors.toArray(new Color[0]);
 		}
-		return get(source, o -> {
-			Colorable colorable = getColorable(o);
 
+		return get(source, obj -> {
+			// ItemType/Item/Slot
+			if (obj instanceof ItemType || obj instanceof Item || obj instanceof Slot) {
+				ItemMeta meta;
+				Material material;
+				if (obj instanceof ItemType) {
+					meta = ((ItemType) obj).getItemMeta();
+					material = ((ItemType) obj).getMaterial();
+				} else if (obj instanceof Item) {
+					meta = ((Item) obj).getItemStack().getItemMeta();
+					material = ((Item) obj).getItemStack().getType();
+				} else {
+					 if (((Slot) obj).getItem() == null)
+						 return null;
+					meta = ((Slot) obj).getItem().getItemMeta();
+					material = ((Slot) obj).getItem().getType();
+				}
+
+				if (meta == null)
+					return null;
+
+				if (meta instanceof LeatherArmorMeta) {
+					LeatherArmorMeta leatherArmorMeta = (LeatherArmorMeta) meta;
+					return ColorRGB.fromBukkitOrRgbColor(leatherArmorMeta.getColor());
+				} else if (meta instanceof MapMeta && MAPS_AND_POTIONS_COLORS) {
+						MapMeta mapMeta = (MapMeta) meta;
+						if (mapMeta.getColor() != null)
```

Comment:
```suggestion
						if (mapMeta.hasColor())
```
You need to use this check per the javadoc of getColor